agent/config/parse.go

```diff
@@ -297,6 +297,28 @@ func parseTaskMetadataThrottles() (int, int) {
 	return steadyStateRate, burstRate
 }
 
+func parseTaskPidsLimit() int {
```

Comment:
To confirm, there is no upper bound for PID limit as long as it is a valid int right?

Reply:
Q. Should we add this new function to parse_linux.go because this PID limit only works for Linux? The same question for type.go. Thanks.

Reply:
My understanding of the "platform variables" struct is that it will be for variables that can never be supported by linux. Since this feature can be supported by windows there's no need to isolate it to a different linux-specific config struct.

Reply:
Seems like the unofficial limit in OCI/containerd is 4194304 (1024*1024*4):
```
% docker run -d --pids-limit 4194304 ubuntu:latest sleep 999
47f4622c595cf0b7672280c62d7483d4d9f5ba69f25c0f40567d4243bd882db3

% docker run -d --pids-limit 4194305 ubuntu:latest sleep 999
f436b71d39c6fc7afbf408aa2fb64ddb7863d0c6e0b58dbcc32a4273346bfbb5
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error setting cgroup config for procHooks process: failed to write "4194305": write /sys/fs/cgroup/pids/docker/f436b71d39c6fc7afbf408aa2fb64ddb7863d0c6e0b58dbcc32a4273346bfbb5/pids.max: invalid argument: unknown.
```
I will go with this limit in our code too, good callout, thanks!!